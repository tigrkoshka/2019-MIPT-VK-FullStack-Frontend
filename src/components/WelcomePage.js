import React from 'react'
import { Link } from 'react-router-dom'
import welcomePageStyles from '../styles/welcomePageStyles.module.scss'

class WelcomePage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      message: (
        <div className={welcomePageStyles.content}>
          Thank you for choosing Hummingbird!
          <br />
          Enter your tag and password.
          <br />
          Then tap anywhere to continue.
        </div>
      ),
      tag: '',
      password: '',
    }

    this.handleTagChange = this.handleTagChange.bind(this)
    this.handlePasswordChange = this.handlePasswordChange.bind(this)
    this.handleTagSubmit = this.handleTagSubmit.bind(this)
    this.handleAuth = this.handleAuth.bind(this)
    this.passwordInput = React.createRef()
  }

  handleTagChange(event) {
    event.preventDefault()
    this.setState({ tag: event.target.value })
  }

  handlePasswordChange(event) {
    event.preventDefault()
    this.setState({ password: event.target.value })
  }

  handleTagSubmit(event) {
    event.preventDefault()
    if (this.state.password === '') {
      this.passwordInput.current.focus()
    } else {
      this.handleAuth(event)
    }
  }

  handleAuth(event) {
    event.preventDefault()
    fetch(`http://127.0.0.1:8000/users/auth/?tag=${this.state.tag}&password=${this.state.password}`)
      .then((res) => (res.ok ? res.json() : Promise.reject(res)))
      .then(({ id }) => {
        window.location.hash = `#/ChatList/${id}`
      })
      .catch(() => {
        this.setState({
          message: (
            <div className={welcomePageStyles.content}>
              Incorrect tag of password.
              <br />
              Try again.
            </div>
          ),
        })
      })
  }

  render() {
    return (
      <div className={welcomePageStyles.vert} onClick={this.handleAuth}>
        <div className={welcomePageStyles.horiz}>
          <div className={welcomePageStyles.curr_message}>
            {this.state.message}
            <div className={welcomePageStyles.time}>2:39</div>
          </div>
        </div>
        <div className={welcomePageStyles.empty} />
        <div className={welcomePageStyles.horiz}>
          <form onSubmit={this.handleTagSubmit} className={welcomePageStyles.form}>
            <input
              type="text"
              value={this.state.tag}
              placeholder="Enter your tag"
              className={welcomePageStyles.input}
              onChange={this.handleTagChange}
              onClick={(event) => {
                event.stopPropagation()
                return false
              }}
            />
          </form>
        </div>
        <div
          className={welcomePageStyles.empty}
          onClick={(event) => {
            event.stopPropagation()
            return false
          }}
        />
        <div className={welcomePageStyles.horiz}>
          <form onSubmit={this.handleAuth} className={welcomePageStyles.form}>
            <input
              type="password"
              value={this.state.password}
              placeholder="Enter your password"
              className={welcomePageStyles.input}
              onChange={this.handlePasswordChange}
              onClick={(event) => {
                event.stopPropagation()
                return false
              }}
              ref={this.passwordInput}
            />
          </form>
        </div>
        <div
          className={welcomePageStyles.empty}
          onClick={(event) => {
            event.stopPropagation()
            return false
          }}
        />
        <Link
          to="/CreateUser"
          className={welcomePageStyles.create}
          onClick={(event) => {
            event.stopPropagation()
            return false
          }}
        >
          Don&apos;t have an account? Become a hummingbird!
        </Link>
      </div>
    )
  }
}

export default WelcomePage
