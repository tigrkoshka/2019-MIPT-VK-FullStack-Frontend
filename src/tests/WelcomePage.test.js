/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import WelcomePage from '../components/WelcomePage'

describe('Welcome page renders and acts correctly', () => {
  let component

  beforeEach(() => {
    component = renderer
      .create(
        <MemoryRouter>
          <WelcomePage />
        </MemoryRouter>,
      )
      .toJSON()
  })

  it('Should render normally', () => {
    expect(component).toMatchSnapshot()
  })
})
