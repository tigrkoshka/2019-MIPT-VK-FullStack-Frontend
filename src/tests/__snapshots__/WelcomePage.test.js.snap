// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome page renders and acts correctly Should render normally 1`] = `
<div
  className="vert"
  onClick={[Function]}
>
  <div
    className="horiz"
  >
    <div
      className="curr_message"
    >
      <div
        className="content"
      >
        Thank you for choosing Hummingbird!
        <br />
        Enter your tag and password.
        <br />
        Then tap anywhere to continue.
      </div>
      <div
        className="time"
      >
        2:39
      </div>
    </div>
  </div>
  <div
    className="empty"
  />
  <div
    className="horiz"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Enter your tag"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    className="empty"
    onClick={[Function]}
  />
  <div
    className="horiz"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="input"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Enter your password"
        type="password"
        value=""
      />
    </form>
  </div>
  <div
    className="empty"
    onClick={[Function]}
  />
  <a
    className="create"
    href="/CreateUser"
    onClick={[Function]}
  >
    Don't have an account? Become a hummingbird!
  </a>
</div>
`;
